commit bd3b9ac829ab8a0ce4e13c904592c5be70bd6982
Author: Alexandre TORGUE <alexandre.torgue@st.com>
Date:   Tue May 30 16:43:04 2017 +0200

    pinctrl: stm32: Fix bad function call
    
    commit b7c747d4627462f25b3daabf49c18895a6722faa upstream.
    
    In stm32_pconf_parse_conf function, stm32_pmx_gpio_set_direction is
    called with wrong parameter value. Indeed, using NULL value for range
    will raise an oops.
    
    Fixes: aceb16dc2da5 ("pinctrl: Add STM32 MCUs support")
    Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: Alexandre TORGUE <alexandre.torgue@st.com>
    Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/pinctrl/stm32/pinctrl-stm32.c b/drivers/pinctrl/stm32/pinctrl-stm32.c
index efc4371..a36fd4b 100644
--- a/drivers/pinctrl/stm32/pinctrl-stm32.c
+++ b/drivers/pinctrl/stm32/pinctrl-stm32.c
@@ -798,7 +798,7 @@ static int stm32_pconf_parse_conf(struct pinctrl_dev *pctldev,
 		break;
 	case PIN_CONFIG_OUTPUT:
 		__stm32_gpio_set(bank, offset, arg);
-		ret = stm32_pmx_gpio_set_direction(pctldev, NULL, pin, false);
+		ret = stm32_pmx_gpio_set_direction(pctldev, range, pin, false);
 		break;
 	default:
 		ret = -EINVAL;
